backend/src/libs/packages/socket/socket.service.ts

```diff
@@ -42,6 +104,75 @@ class SocketService {
           this.geolocationCacheService.setCache(driverId, latLng);
         },
       );
+
+      socket.on(
+        ServerSocketEvent.START_SHIFT,
+        async (
+          payload: ServerSocketEventParameter[typeof ServerSocketEvent.START_SHIFT],
+          callback: (
+            status: ValueOf<typeof SocketResponseStatus>,
+            message?: string,
+          ) => void,
+        ): Promise<void> => {
+          if (!user) {
+            return callback(
+              SocketResponseStatus.BAD_EMIT,
+              SocketError.NOT_AUTHORIZED,
+            );
+          }
+
+          const { truckId } = payload;
+          const isTruckNotAvailable =
+            await this.truckService.checkIsNotAvailableById(truckId);
+
+          if (isTruckNotAvailable) {
+            return callback(
+              SocketResponseStatus.BAD_EMIT,
+              SocketError.TRUCK_NOT_AVAILABLE,
+            );
+          }
+
+          const timer = setTimeout(
+            (user: UserEntityObjectWithGroupT) => {
+              socket.emit(ClientSocketEvent.DRIVER_TIMED_OUT);
+              void socketEndShift({
+                io: this.io,
+                startedShiftsStore,
+                shiftService,
+                truckService,
+                user,
+              });
+            },
+            HOUR_IN_MS,
+            user,
+          );
+
+          await socketChooseTruck(truckId, this.truckService, this.io);
+
+          await createShift({
+            startedShiftsStore,
+            user,
+            truckId,
+            shiftService: this.shiftService,
+            startedShift: { socket, timer },
+          });
+
+          callback(SocketResponseStatus.OK);
+        },
+      );
+
+      socket.on(ServerSocketEvent.END_SHIFT, async () => {
+        if (!user || !startedShiftsStore.has(user.id)) {
+          return;
+        }
+        await socketEndShift({
+          io: this.io,
+          startedShiftsStore,
+          truckService,
+          shiftService,
+          user,
+        });
+      });
     });
```

Comment:
It seems to me it's better to inject socket service in truckService,  shiftService, userService, where we need and keep socket service as thin as possible